### DISCUSSION TITLE
separate api client from main

### DISCUSSION THREAD

**Discussion by abcnow** [serverpod/serverpod#1477] [General]
I think it will be great if we can separate Api Client from Main.. something like this here:

https://github.com/tylersavery/full-stack-dart/blob/main/listie2_flutter/lib/src/utils/api_client.dart

abstract class ApiClient {
  late SessionManager sessionManager;
  late Client client;

  Future<void> init();
}

class ApiClientImplementation extends ApiClient {
  @override
  Future<void> init() async {
    client = Client(
      'https://api.name.online/',
      authenticationKeyManager: FlutterAuthenticationKeyManager(),
    )..connectivityMonitor = FlutterConnectivityMonitor();

    sessionManager = SessionManager(caller: client.modules.auth);
    await sessionManager.initialize();
  }
}

**Comment by vlidholt** [comment]
Can you describe a bit more what you want to do, and what your use case is?
